transitions/registration.js

```diff
@@ -217,21 +216,29 @@ module.exports = {
                 if (self.isBoolExprFalse(obj, event.bool_expr)) {
                     return;
                 }
-                var options = { db: db, audit: audit, doc: doc };
-                if (event.params) {
-                    // params setting get sent as array
+                var options = { db: db, audit: audit, doc: doc, registrationConfig: registrationConfig };
+
+                if (typeof event.params === 'string') {
+                    // params setting can get sent as string to convert into an array
                     options.params = event.params.split(',');
+                } else if (typeof event.params === 'object') {
```

Comment:
OK. I cannot think of a clean way of doing this, but maybe that's the Friday in me speaking. Let's not push it. Let's talk Monday. Ideally what I want, is for people to be able to add actual JSON (not a String we all have a secret handshake to agree is JSON). If that has to be a different property, I'd suggest we think of a better name than "param" and then deprecate its usage in favour of this one.